### PR TITLE
changing custom-properties to ini_setting

### PR DESCRIPTION
**Pull request by nikolas-hermanns**
There is no need to have the template of the whole custom.properties. It is enough to change only the lines wanted.

### CODE REVIEW

**Comment by nikolas-hermanns** [comment]
@dfarrell07

**Comment by nikolas-hermanns** [comment]
@dfarrell07 here is the pull request

File: manifests/config.pp [hunk 1]
```diff
@@ -31,14 +31,35 @@
...
     ensure  => file,
-    path    => '/opt/opendaylight/etc/custom.properties',
+    path    => $custom_properties,
     # Set user:group owners
     owner   => 'odl',
     group   => 'odl',
-    # Use a template to populate the content
-    content => template('opendaylight/custom.properties.erb'),
+  }
+  if $opendaylight::enable_l3 == 'yes' or $opendaylight::enable_l3 == true {
```
Comment (trozet) [inline manifests/config.pp]: This actually doesnt do anything anymore with new netvirt.  We can remove the enable_l3 parameter.  We should actually cut a stable/boron branch, and you can remove this on master.

Comment (trozet) [inline manifests/config.pp]: Also consider using str2bool() for cheking values that can be yes or true

File: manifests/config.pp [hunk 2]
```diff
@@ -31,14 +31,35 @@
...
+    ini_setting {'ovsdb.l3.fwd.enabled':
+      ensure => present,
+      setting => 'ovsdb.l3.fwd.enabled',
+      value => 'yes',
+      path => $custom_properties
+    }
+    ini_setting {'ovsdb.l3.arp.responder.disable':
```
Comment (trozet) [inline manifests/config.pp]: iirc this also does nothing anymore.  Need to double check with Sam Hague.

File: manifests/config.pp (hunk 1 above)
Comment (nikolas-hermanns) [inline manifests/config.pp]: ah ok so odl will always be the l3 router now?

**Comment by nikolas-hermanns** [comment]
I think that is a good idea. do we have a list of possible configuration for odl so that we can catch most of the things?

Some things are configured in xml files. I have not yet found out how to configure xml files with puppet.

**Comment by dfarrell07** [comment]
> I have not yet found out how to configure xml files with puppet.

Maybe the [xmlfile](https://forge.puppet.com/terc/xmlfile) Puppet module to abstract [Augeas](https://docs.puppet.com/guides/augeas.html).

**Comment by nikolas-hermanns** [comment]
What do you think about that?

**Comment by trozet** [comment]
@nikolas-hermanns this is just failing now because there are leftover spec tests failing.  Can you also remove the spec tests?

**Comment by dfarrell07** [comment]
> @nikolas-hermanns this is just failing now because there are leftover spec tests failing. Can you also remove the spec tests?

+1. I wish there was a good way to amend someone else's patch with GitHub like you can with Gerrit, I'd remove them for you.

**Comment by dfarrell07** [comment]
I fixed the Beaker tests in #126, so we can use them to verify this change.

**Comment by dfarrell07** [comment]
> This actually doesnt do anything anymore with new netvirt. We can remove the enable_l3 parameter. - @trozet 

> ah ok so odl will always be the l3 router now? - @nikolas-hermanns 

So L3 is enabled by default in the new NetVirt? If so, we need to fix the docs.

Latest stable/boron (pre-SR3) etc/custom.properties:

```
# ovsdb can be configured with ml2 to perform l3 forwarding. The config below enables that functionality, which is
# disabled by default.
# ovsdb.l3.fwd.enabled=yes
```

Latest master (pre-Carbon) etc/custom.properties:

```
# ovsdb can be configured with ml2 to perform l3 forwarding. The config below enables that functionality, which is
# disabled by default.
# ovsdb.l3.fwd.enabled=yes
```

Do we still want this knob at all? Does new NetVirt respect it? Can you disable L3 with it, and is that useful enough to justify keeping the param?

**Comment by dfarrell07** [comment]
I have a finished version of this patch in #129. Beaker and rspec tests are passing.

**Comment by shague** [comment]
new netvirt doesn't look at any of the params in the custom.properties. There is no way to disable l3 in the new netvirt. I thought Tim wanted to keep the params around though so that both netvirt versions would be used?

**Comment by dfarrell07** [comment]
> new netvirt doesn't look at any of the params in the custom.properties. There is no way to disable l3 in the new netvirt.

Awesome, very good to know, thanks @shague.

>  thought Tim wanted to keep the params around though so that both netvirt versions would be used?

Tim cut a stable/boron branch that will still have this param. I'm guessing he doesn't want it for Carbon, based on above. @trozet?

**Comment by trozet** [comment]
@dfarrell07 Correct.  This patch should only go into master.

**Comment by dfarrell07** [comment]
@trozet Okay, cool. So I'll go ahead and merge #129 and abandon this.

Thanks all :+1:

**Pull request closed without merging**